packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -493,6 +494,10 @@ export async function activate(context: vscode.ExtensionContext) {
     decorators.showDemoMode();
   }
 
+  // Refresh SObject definitions if there aren't any faux classes
+  const projectPath = vscode.workspace!.workspaceFolders![0].uri.fsPath;
+  await initSObjectDefinitions(projectPath);
```

Comment:
FYI this does not mean wait until the definitions have been completely refreshed, it's just waiting until executeCommand was called (or not called) in the function.

Reply:
Do we need to do wait for that ? `initSObjectDefinitions` currently runs ForceConfigGet (via `workspaceOrgType`) which is running `sfdx force:config:get` which can be very slow depending on how many credentials you have in your local cli.